site/src/main/scala/org/typelevel/sbt/TypelevelSitePlugin.scala

```diff
@@ -71,7 +83,21 @@ object TypelevelSitePlugin extends AutoPlugin {
       )
       .value: @nowarn("cat=other-pure-statement"),
     Laika / sourceDirectories := Seq(mdocOut.value),
-    laikaTheme := tlSiteHeliumConfig.value.build,
+    tlSiteTheme := {
+      TypelevelHeliumTheme(
+        gitHubUserRepo.value.map(_._2),
+        developers.value,
+        version.value,
+        tlSiteApiUrl.value,
+        scmInfo.value.map(_.browseUrl),
+        licenses.value.headOption,
+        tlSiteRelated.value
+      )
+    },
+    tlSiteHeliumConfig := Helium.defaults,
+    laikaTheme := {
+      tlSiteHeliumConfig.value.build.extend(tlSiteTheme.value)
+    },
```

Comment:
This configuration still feels awkward 😕

Reply:
What would you prefer as an alternative here? 

I assume from the user's perspective they can completely ignore the `laikaTheme` setting here (which would need to be documented as they might be tempted to use it by looking at Laika docs) and the remaining 2 are the core Helium features and the Typelevel extra theme features which I feel are best kept separate. For one of the settings you'd add docs to your own site and for the other you could directly point them at the Laika page for theme settings.

The only way you could create a unified theme config API would be if the base theme would be a fairly small feature set that would enable you to re-expose its config API in your own theme API. But the Helium API is huge and replicating it in an extension is not feasible/desirable.

Reply:
Now looking at the implementation I'd recommend not to completely hide the core Helium config inside the `TypelevelHeliumExtensions` class. The hidden defaults would override anything users would set in `tlSiteHeliumConfig`. You also kind of create the Helium theme twice (which would also lead to some things executing twice). I feel like it's more convenient and better separated if users could write something like this:

```scala
tlSiteHeliumBase = TypeLevelHeliumBase.defaults
  .withFoo
  .withBar...

tlSiteHeliumExtensions = TypeLevelHeliumExtensions.defaults
  .withFoo
  .withBar...
```

Where the first setting is the core Helium API as documented in the Laika manual, only that the defaults are already richer than the builtin `Helium.defaults` (essentially the `base` value currently hidden inside the extension impl) and the second setting is your own API with functionality not covered by core Helium.

This is certainly not ideal, but maybe the best option given that wrapping and re-exposing the extensive Helium API is not feasible?

Reply:
> the hidden defaults would override anything users would set in `tlSiteHeliumConfig`. You also kind of create the Helium theme twice (which would also lead to some things executing twice).

Thanks, yes, those were exactly the problems I was concerned about. Actually this change was not backwards-compatible because of the overriding thing.

> This is certainly not ideal, but maybe the best option given that wrapping and re-exposing the extensive Helium API is not feasible?

You are right, I was trying to avoid having two separate settings for the base config and extensions. But it seems unavoidable, so I agree it is probably the best option.

> I assume from the user's perspective they can completely ignore the `laikaTheme` setting here (which would need to be documented as they might be tempted to use it

Right, well it depends whether the user is a Typelevel project or not. In fact, I don't expect most Typelevel projects to care about these settings at all. Only a few projects might need to make some very minor customization, except the sbt-http4s-org plugin which will add its own extensions.

https://github.com/typelevel/sbt-typelevel/blob/791f9a905ba0d54e800759a39f8609ecab62e6c7/build.sbt#L152-L156

https://github.com/http4s/http4s-dom/blob/cce807dfec6754e89b1eae73b306401b5b3997e6/build.sbt#L200-L204

 Non-typelevel projects can ignore these and make their own theme following the Laika manual.

Reply:
Additional note: it might also be worthwhile to check whether the things you do in your extension settings are generic enough to be offered by the core Helium API, but that would be a thing for 0.19 and not something you can plan with for this PR.